zaas-client/src/main/java/org/zowe/apiml/zaasclient/client/HttpsClient.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.zaasclient.client;
+
+import lombok.AllArgsConstructor;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.zowe.apiml.zaasclient.config.ConfigProperties;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.*;
+import java.security.cert.CertificateException;
+
+@AllArgsConstructor
+public class HttpsClient implements Closeable {
+    private CloseableHttpClient closeableHttpsClient;
+    private final String keyStorePassword;
+    private final String trustStorePassword;
+    private final String keyStoreType;
+    private final String trustStoreType;
+    private final RequestConfig requestConfig;
+    private String trustStorePath;
+    private String keyStorePath;
+
+    public HttpsClient(ConfigProperties configProperties) {
+        this.requestConfig = this.buildCustomRequestConfig();
+        this.keyStorePath = configProperties.getKeyStorePath();
+        this.keyStorePassword = configProperties.getKeyStorePassword();
+        this.keyStoreType = configProperties.getKeyStoreType();
+        this.trustStorePath = configProperties.getTrustStorePath();
+        this.trustStorePassword = configProperties.getTrustStorePassword();
+        this.trustStoreType = configProperties.getTrustStoreType();
+    }
+
+
+    public CloseableHttpClient getHttpsClientWithTrustStore()
+        throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException, KeyManagementException {
+        SSLContext sslContext;
+        TrustManagerFactory tmf = null;
+
+        if (trustStorePath != null) {
+            tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+            File trustFile = new File(trustStorePath);
+            trustStore.load(new FileInputStream(trustFile), trustStorePassword.toCharArray());
+            tmf.init(trustStore);
+        }
```

Comment:
This block is repeating, shouldn't it be as separate method?
If path is null, how would user know? Does he need to know?

Reply:
Thanks for the feedback. We are working on error handling(based on the reviews received from Petr). Sooner you will get latest updates in the branch.